tests/functional/idempotency/test_idempotency.py

```diff
@@ -638,3 +641,44 @@ def test_delete_from_cache_when_empty(persistence_store):
     except KeyError:
         # THEN we should not get a KeyError
         pytest.fail("KeyError should not happen")
+
+
+def test_is_missing_idempotency_key():
+    # GIVEN None THEN is_missing_idempotency_key is True
+    assert BasePersistenceLayer.is_missing_idempotency_key(None)
+    # GIVEN a list of Nones THEN is_missing_idempotency_key is True
+    assert BasePersistenceLayer.is_missing_idempotency_key([None, None])
+    # GIVEN a list of all not None THEN is_missing_idempotency_key is false
+    assert BasePersistenceLayer.is_missing_idempotency_key([None, "Value"]) is False
+    # GIVEN a str THEN is_missing_idempotency_key is false
+    assert BasePersistenceLayer.is_missing_idempotency_key("Value") is False
+
```

Comment:
```suggestion
    assert BasePersistenceLayer.is_missing_idempotency_key("Value") is False

    assert BasePersistenceLayer.is_missing_idempotency_key(()) is False
    # GIVEN an empty dictionary THEN is_missing_idempotency_key is false
    assert BasePersistenceLayer.is_missing_idempotency_key({}) is False
```

Add tests for non-lists but iterable too